git/git_edit.nu

```diff
@@ -0,0 +1,33 @@
+# Custom command to conveniently edit commits in the history.
+#
+# Use with:
+# `use <repo_root>/git/git_edit.nu`
+#
+# `git edit` starts an interactive rebase to edit a commit in the history, which
+# can be selected from a complection list. An example completion list may look
+# like this:
+#
+# >| git edit
+# 2024d52  upgrade to nu 0.77 (#413)
+# 10bde06  Fix since last release script (#412)
+# 18deb3d  Add `nu-cmd-lang` to `nu_release.nu` (#411)
+# e95f609  Updated conda script to default to base if no arg given (#409)
+# 34235ef  Replace deprecated `str collect` command (#410)
+# 06dd125  Revert "conda activate base if env_name not given (#400)" (#408)
+# ...
+
+# Helper command to get commit of this branch only list to edit
+def "nu-complete git commits" [] {
+  ^git log --pretty="%h %s" | lines | parse "{value} {description}"
+}
+
+# Edit a commit by selecting it as only commit to edit in an interactive rebase
+# run non-interactively.
+export def "git edit" [
+    commit: string@"nu-complete git commits"
+] {
+    # We use `sed` to replace `pick` with `edit`.
+    # `0,/pick/` restricts the sustitution to the range from the beginning until
+    # the first occurence of `pick`.
+    GIT_SEQUENCE_EDITOR="sed -i -e '0,/pick/{s/pick/edit/}'" ^git rebase -i $"($commit)^1"
```

Comment:
If we could change this `sed` command to something more nushell friendly, this could work cross-platform. Maybe `str replace` would help?

Reply:
Thanks for the suggestion @fdncred ! In theory, a simple `str replace "pick" "edit"` would do as it replaces only the first occurrence as we want.

However, I could not get `git` to invoke `nu` in the right way :thinking: On my machine, I do not (yet?) use `nu` as login shell as I am not sure about the impact on the overall system. Thus the command in `GIT_SEQUENCE_EDITOR` is invoked in bash. Could you help me out on how to invoke the equivalent of ` ... | str replace "pick" "edit"` in the form of `nu -c ...`?

Reply:
I think this is what you're looking for. You can test this out in the repl.
```
"pick" | nu --stdin -c 'str replace pick edit'
```

Reply:
Hm would it be OK then maybe to move forward with this in its "unix-only" form and leave it up for someone else to update it to a portable nushell-only based approach :thinking:

Reply:
I'm not a fan of having unix-only stuff.

Reply:
Sure, nushell is a cross-platform shell which means we try to support MacOS, Linux, and Windows in everything we do. So, it's important that some of our core scripts are supported cross-platform like git and cargo, for instance. We don't want to have people use some of these fundamental scripts and they just don't work because they're on Windows (or other platforms). At some point we may include git, cargo, and maybe other scripts in the nushell distribution, so we need to be careful with the cross-platform nature of them. 

This repo is also a proving ground for missing features that may need to be included in nushell. So, if there's something nushell needs to do that isn't included in nushell yet, we can notice it here as the community tries out things before nushell reaches 1.0.

Reply:
I understand now, although it seems a different vision than what the description of the repository sells:

> A place to share Nushell scripts with each other

I thought by the words of it, this was community-centered and not future-of-nushell-centered. I think it can be worded better.

Reply:
It doesn't have to be either or, it can be both. This is community-centered, but you have to remember, you don't represent all of the community. This is something that we struggle with sometimes because there are Linux zealots, MacOS zealots, and Windows zealots that preach to us how we should do things and we try really hard to have a balance and do our best to support all three of these excellently. It's not easy, especially when the team is 100% volunteer.

Reply:
If there hasn't been an issue created in the nushell repo for whatever we need here to move forward, then please create one.

Reply:
> an issue created in the nushell repo for whatever we need here to move forward

I don't think it has to be one, what I don't seem to understand is that **if this is for the community**, the scripts don't have to run in **every device of every person in the community.** A community, like the Linux one, have hundreds of different operational systems, thousands of different distros, hundreds of thousands of different software, many conflicting with many others, **it doesn't make them an unique special community, it makes them a community that even with its own separate groups is part of the larger Linux community**.

A script, in whatever script language you can think of, has its own limitations, sometimes it will need dependencies (libraries, plugins, external software, etc) to do a task in a interoperable manner. It's not necessarily a bad thing, many times it is just what needs to be done to do what we want it to do.

What is the point you're trying to give here? You don't like an external software, therefore it breaks the community guideline? It's not like `sed` cannot be installed in MacOS or Windows. It's not like `sed` is more terrific than `pueue`. It's just software. And what's a better place to share software than in a place worded to do exactly that:

> A place to share Nushell scripts with each other